panic: opBmap when reading from mount on FreeBSD 12.1-RELEASE-p2

<!--

Welcome! - We kindly ask that you:

  1. Fill out the issue template below - not doing so needs a good reason.
  2. Use the forum if you have a question rather than a bug or feature request.

The forum is at: https://forum.restic.net

NOTE: Not filling out the issue template needs a good reason, as otherwise it
may take a lot longer to find the problem, not to mention it can take up a lot
more time which can otherwise be spent on development. Please also take the
time to help us debug the issue by collecting relevant information, even if
it doesn't seem to be relevant to you. Thanks!

The forum is a better place for questions about restic or general suggestions
and topics, e.g. usage or documentation questions! This issue tracker is mainly
for tracking bugs and feature requests directly relating to the development of
the software itself, rather than the project.

Thanks for understanding, and for contributing to the project!

-->


Output of `restic version`
--------------------------
restic 0.9.6 compiled with go1.14 on freebsd/amd64

How did you run restic exactly?
-------------------------------
restic -r archive mount mnt

<!--
This section should include at least:

 * The complete command line and any environment variables you used to
   configure restic's backend access. Make sure to replace sensitive values!

 * The output of the commands, what restic prints gives may give us much
   information to diagnose the problem!
-->

What backend/server/service did you use to store the repository?
----------------------------------------------------------------
Local repository created on Windows 10 and then copied to FreeBSD.


Expected behavior
-----------------
When mounting the repository with
`restic -r archive mount mnt
`
then
`cp -R mnt/snapshots/latest/d out/`
should copy d from latest snapshot to out.
<!--
Describe what you'd like restic to do differently.
-->

Actual behavior
---------------
Some files are copied and then restic mount process fails with messages:

restic -r excel mount mnt
enter password for repository: 
repository 8d924547 opened successfully, password is correct
Now serving the repository at mnt
When finished, quit with Ctrl-c or umount the mountpoint.
panic: opBmap

goroutine 1 [running]:
bazil.org/fuse.(*Conn).ReadRequest(0xc00007aba0, 0xdbf928, 0xc003a000c0, 0xed46c0, 0xc0016f3740)
	/wrkdirs/usr/ports/sysutils/restic/work/restic-0.9.6/vendor/bazil.org/fuse/fuse.go:1007 +0x5056
bazil.org/fuse/fs.(*Server).Serve(0xc003a000c0, 0xec46c0, 0xc003a00000, 0x0, 0x0)
	/wrkdirs/usr/ports/sysutils/restic/work/restic-0.9.6/vendor/bazil.org/fuse/fs/serve.go:414 +0x36e
bazil.org/fuse/fs.Serve(0xc00007aba0, 0xec46c0, 0xc003a00000, 0x0, 0x0)
	/wrkdirs/usr/ports/sysutils/restic/work/restic-0.9.6/vendor/bazil.org/fuse/fs/serve.go:435 +0xb9
main.mount(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xd8c8ef, ...)
	/wrkdirs/usr/ports/sysutils/restic/work/restic-0.9.6/cmd/restic/cmd_mount.go:155 +0x545
main.runMount(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xd8c8ef, ...)
	/wrkdirs/usr/ports/sysutils/restic/work/restic-0.9.6/cmd/restic/cmd_mount.go:192 +0x175
main.glob..func16(0x1463780, 0xc00044c450, 0x1, 0x3, 0x0, 0x0)
	/wrkdirs/usr/ports/sysutils/restic/work/restic-0.9.6/cmd/restic/cmd_mount.go:50 +0xbc
github.com/spf13/cobra.(*Command).execute(0x1463780, 0xc0000af4c0, 0x3, 0x4, 0x1463780, 0xc0000af4c0)
	/wrkdirs/usr/ports/sysutils/restic/work/restic-0.9.6/vendor/github.com/spf13/cobra/command.go:762 +0x453
github.com/spf13/cobra.(*Command).ExecuteC(0x1464a80, 0xc00038aed0, 0x4, 0x10)
	/wrkdirs/usr/ports/sysutils/restic/work/restic-0.9.6/vendor/github.com/spf13/cobra/command.go:852 +0x2ea
github.com/spf13/cobra.(*Command).Execute(...)
	/wrkdirs/usr/ports/sysutils/restic/work/restic-0.9.6/vendor/github.com/spf13/cobra/command.go:800
main.main()
	/wrkdirs/usr/ports/sysutils/restic/work/restic-0.9.6/cmd/restic/main.go:86 +0x3f

<!--
In this section, please try to concentrate on observations, so only describe
what you observed directly.
-->

Steps to reproduce the behavior
-------------------------------
Already explained.
<!--
The more time you spend describing an easy way to reproduce the behavior (if
this is possible), the easier it is for the project developers to fix it!
-->

Do you have any idea what may have caused this?
-----------------------------------------------
Seems to be similar to
[https://github.com/rclone/rclone/issues/3697]()


Do you have an idea how to solve the issue?
-------------------------------------------
Already fixed in bazil/fuse?
[https://github.com/bazil/fuse/issues/221](url)


Did restic help you today? Did it make you happy in any way?
------------------------------------------------------------
On linux  `restic mount` works with `rsync` without problems.
<!--
Answering this question is not required, but if you have anything positive to share, please do so here!
Sometimes we get tired of reading bug reports all day and a little positive end note does wonders.
Idea by Joey Hess, https://joeyh.name/blog/entry/two_holiday_stories/
-->

--- a/cmd/restic/cmd_mount.go
+++ b/cmd/restic/cmd_mount.go
@@ -59,7 +59,6 @@ Exit status is 0 if the command was successful, and non-zero if there was any er
 // MountOptions collects all options for the mount command.
 type MountOptions struct {
 	OwnerRoot            bool
-	AllowRoot            bool
 	AllowOther           bool
 	NoDefaultPermissions bool
 	Hosts                []string
@@ -75,7 +74,6 @@ func init() {
 
 	mountFlags := cmdMount.Flags()
 	mountFlags.BoolVar(&mountOptions.OwnerRoot, "owner-root", false, "use 'root' as the owner of files and dirs")
-	mountFlags.BoolVar(&mountOptions.AllowRoot, "allow-root", false, "allow root user to access the data in the mounted directory")
 	mountFlags.BoolVar(&mountOptions.AllowOther, "allow-other", false, "allow other users to access the data in the mounted directory")
 	mountFlags.BoolVar(&mountOptions.NoDefaultPermissions, "no-default-permissions", false, "for 'allow-other', ignore Unix permissions and allow users to read all snapshot files")
 
@@ -117,10 +115,6 @@ func mount(opts MountOptions, gopts GlobalOptions, mountpoint string) error {
 	mountOptions := []systemFuse.MountOption{
 		systemFuse.ReadOnly(),
 		systemFuse.FSName("restic"),
-	}
-
-	if opts.AllowRoot {
-		mountOptions = append(mountOptions, systemFuse.AllowRoot())
 	}
 
 	if opts.AllowOther {

--- a/doc/bash-completion.sh
+++ b/doc/bash-completion.sh
@@ -1026,8 +1026,6 @@ _restic_mount()
 
     flags+=("--allow-other")
     local_nonpersistent_flags+=("--allow-other")
-    flags+=("--allow-root")
-    local_nonpersistent_flags+=("--allow-root")
     flags+=("--help")
     flags+=("-h")
     local_nonpersistent_flags+=("--help")

--- a/doc/man/restic-mount.1
+++ b/doc/man/restic-mount.1
@@ -57,10 +57,6 @@ For details please see the documentation for time.Format() at:
     allow other users to access the data in the mounted directory
 
 .PP
-\fB\-\-allow\-root\fP[=false]
-    allow root user to access the data in the mounted directory
-
-.PP
 \fB\-h\fP, \fB\-\-help\fP[=false]
     help for mount
 

--- a/go.mod
+++ b/go.mod
@@ -1,7 +1,7 @@
 module github.com/restic/restic
 
 require (
-	bazil.org/fuse v0.0.0-20180421153158-65cc252bf669
+	bazil.org/fuse v0.0.0-20191225072544-27e78e7d88df
 	cloud.google.com/go v0.37.4 // indirect
 	github.com/Azure/azure-sdk-for-go v27.3.0+incompatible
 	github.com/Azure/go-autorest/autorest v0.9.2 // indirect
@@ -37,7 +37,7 @@ require (
 	golang.org/x/net v0.0.0-20190522155817-f3200d17e092
 	golang.org/x/oauth2 v0.0.0-20190402181905-9f3314589c9a
 	golang.org/x/sync v0.0.0-20190423024810-112230192c58
-	golang.org/x/sys v0.0.0-20190422165155-953cdadca894
+	golang.org/x/sys v0.0.0-20191210023423-ac6580df4449
 	golang.org/x/text v0.3.1-0.20180807135948-17ff2d5776d2
 	google.golang.org/api v0.3.2
 	google.golang.org/appengine v1.5.0 // indirect

